pil_analyzer/src/condenser.rs

```diff
@@ -77,14 +74,14 @@ pub fn condense<T: FieldElement>(
 }
 
 pub struct Condenser<T> {
-    pub symbols: HashMap<String, Symbol>,
+    pub symbols: HashMap<String, (Symbol, Option<FunctionValueDefinition<T>>)>,
```

Comment:
Can you add a comment explaining the data structure?

Reply:
I still don't understand this type, also with the comment. What's a `Option<FunctionValueDefinition<T>>` ?

Reply:
Maybe not too important

Reply:
It's an `Option` because the symbol can just be a declaration without a value. `FunctionValueDefinition` is an enum that contains all the different ways to define a column:
```
col fixed x(i) { ... };
col fixed x = [1,2,3]* + [0];
col fixed x = |i| { ... }; // or col fixed x = y;
```